dice_ml/data_interfaces/public_data_interface.py

```diff
@@ -404,10 +404,10 @@ def get_decimal_precisions(self, output_type="list"):
         precisions_dict = defaultdict(int)
         precisions = [0] * len(self.feature_names)
         for ix, col in enumerate(self.continuous_feature_names):
-            if ((self.continuous_features_precision is not None) and (col in self.continuous_features_precision)):
+            if (self.continuous_features_precision is not None) and (col in self.continuous_features_precision):
                 precisions[ix] = self.continuous_features_precision[col]
                 precisions_dict[col] = self.continuous_features_precision[col]
-            elif ((self.data_df[col].dtype == np.float32) or (self.data_df[col].dtype == np.float64)):
+            elif (self.data_df[col].dtype == np.float32) or (self.data_df[col].dtype == np.float64):
                 modes = self.data_df[col].mode()
                 maxp = len(str(modes[0]).split('.')[1])  # maxp stores the maximum precision of the modes
                 for mx in range(len(modes)):
```

Comment:
Probably you could drop these parenthesis as well. See the operator precedence order in the list at the link https://data-flair.training/blogs/python-operator-precedence/#:~:text=Python%20Operator%20Precedence%20%E2%80%93%20Short%20Circuiting%20Python%20always,the%20second%20operand%20only%20until%20it%20is%20needed.